pkg/synthetictests/kubelet.go

```diff
@@ -363,3 +364,70 @@ func testSystemDTimeout(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
 	// write a passing test to trigger detection of this issue as a flake. Doing this first to try to see how frequent the issue actually is
 	return []*ginkgo.JUnitTestCase{failure, success}
 }
+
+var errImagePullTimeoutRE = regexp.MustCompile("ErrImagePull.*read: connection timed out")
+var errImagePullGenericRE = regexp.MustCompile("ErrImagePull")
+
+func testErrImagePullConnTimeoutOpenShiftNamespaces(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull read connection timeout in openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullTimeoutRE, nil, true, events)
+}
+
+func testErrImagePullConnTimeout(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull read connection timeout in non-openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullTimeoutRE, nil, false, events)
+}
+
+func testErrImagePullGenericOpenShiftNamespaces(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull in openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullGenericRE, errImagePullTimeoutRE, true, events)
+}
+
+func testErrImagePullGeneric(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-node] should not encounter ErrImagePull in non-openshift namespace pods"
+	return buildTestsFailIfRegexMatch(testName, errImagePullGenericRE, errImagePullTimeoutRE, false, events)
+}
+
+func buildTestsFailIfRegexMatch(testName string, matchRE, dontMatchRE *regexp.Regexp, inOCPNS bool, events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
```

Comment:
bools in APIs like this age poorly.  in the future, choose something like a namespace selector function or a struct that contains these options.

Reply:
Replacing with an enum